crypto/scrypt/scrypt.go

```diff
@@ -0,0 +1,52 @@
+package scrypt
+
+import (
+	"crypto/rand"
+
+	"golang.org/x/crypto/scrypt"
+)
+
+type Scrypt struct {
+	SaltLen int `json:"saltlen,omitempty" toml:"saltlen"`
+	N       int `json:"n,omitempty" toml:"n"`
+	R       int `json:"r,omitempty" toml:"r"`
+	P       int `json:"p,omitempty" toml:"p"`
+}
+
+const (
+	defaultSaltLen = 16
+	defaultN       = 131072 // 2^17
```

Comment:
You can play around with the n, r and p settings. The current default config takes about 800ms for key derivation on my hardware.